parcels/particlesets/iterators.py

```diff
@@ -63,6 +72,12 @@ def update_next_dt(self, next_dt=None):
     def delete(self):
         # == OBJECTION CK: the actual operation, which is the particle's state, shall be done by the particle. So, == #
         # == this function should just forward the delete-call to the particle in question.                        == #
+        # == RESPONSE RB: There might be a problem with that in some cases. As we discussed, the ParticleAccessor  == #
+        # == basically acts like a shell around the particle (data), providing uniform access regardless of the    == #
+        # == underlying datastructure. The SOA approach, for example, does not use a particle class on which       == #
+        # == functions can be defined. So it makes sense, I think, to implement any function that requires more    == #
+        # == logic than just setting a property on the Accessor level. The state-thing is not really a good        == #
+        # == example, although the delete-alias may be useful over just treating it as a property.                 == #
```

Comment:
The issue is that if we, e.g., change state codes, or need to differentiate and check properties of a particle for execution the right functions, we wanna do that once, on the particle level, and then have this affecting all ParticleSet implementations. I do not wanna go over each specific particle set implementation when changing a particular particle behaviour that is common to all kinds of particle sets.

Reply:
I sort of see your point. There is however no way around some code duplication in this way: as whatever you implement on the particle level has to be recreated somewhere else for datastructures that do not use particle objects (like the SoA approach). So then you end up with a bit of a halfway-solution, where for most datastructures the Accessor just passes the function call along, while for at least one it actually implements something by itself.

Reply:
agreed